File: dotnet/backend/DTO/Destination/DestinationDTO.cs

namespace Etour_Backend_dotnet.DTO.Destination;

public class DestinationDTO
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string ImagePath { get; set; } = string.Empty;
    public string BestTimeToVisit { get; set; } = string.Empty;
    public string Temperature { get; set; } = string.Empty;
    public List<string> Features { get; set; } = new();
    public List<ReviewDTO> Reviews { get; set; } = new();
}

public class ReviewDTO
{
    public string Name { get; set; } = string.Empty;
    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
}

File: java/backend/src/main/java/com/etour/app/dto/DestinationDTO.java
package com.etour.app.dto;
import java.util.List;
public class DestinationDTO {
    private int id;
    private String name;
    private String bestTimeToVisit;
    private String temperature;
    private List<String> features;
    private List<ReviewDTO> reviews;
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getBestTimeToVisit() { return bestTimeToVisit; }
    public void setBestTimeToVisit(String bestTimeToVisit) { this.bestTimeToVisit = bestTimeToVisit; }
    public String getTemperature() { return temperature; }
    public void setTemperature(String temperature) { this.temperature = temperature; }
    public List<String> getFeatures() { return features; }
    public void setFeatures(List<String> features) { this.features = features; }
    public List<ReviewDTO> getReviews() { return reviews; }
    public void setReviews(List<ReviewDTO> reviews) { this.reviews = reviews; }
    public static class ReviewDTO {
        private String name;
        private int rating;
        private String comment;
        public String getName() { return name; }
        public void setName(String name) { this.name = name; }
        public int getRating() { return rating; }
        public void setRating(int rating) { this.rating = rating; }
        public String getComment() { return comment; }
        public void setComment(String comment) { this.comment = comment; }
    }
}